src/ExpressionEvaluator/Core/Source/ExpressionCompiler/DkmUtilities.cs

```diff
@@ -30,7 +30,11 @@ private static IEnumerable<DkmClrModuleInstance> GetModulesInAppDomain(this DkmC
 
             var appDomainId = appDomain.Id;
             return runtime.GetModuleInstances().
-                Cast<DkmClrModuleInstance>().
+                // Ignore module instances that are not DkmClrModuleInstance.
+                // Specifically, ignore DkmClrNcContainerModuleInstance.
+                // (The embedded DkmClrNcModuleInstance modules will
+                // be included in GetModuleInstances and will be used.)
```

Comment:
Updated comment:
// GetModuleInstances() may include instances of DkmClrNcContainerModuleInstance
// which are containers of managed module instances (see GetEmbeddedModules())
// but not managed modules themselves. Since GetModuleInstances() will include the
// embedded modules as well, we can simply ignore DkmClrNcContainerModuleInstances.